### PR TITLE
Update docker/docker, containerd, runc, and some dependencies

### PR DESCRIPTION
**Pull request by thaJeztah**
closes https://github.com/docker/cli/pull/1753

Updates docker/docker and dependencies:

Full diffs:

- https://github.com/docker/docker/compare/8aca18d631f3f72d4c6e3dc01b6e5d468ad941b8...827cb09f87964ed38b46502f22a585f2ed4a78e1
- golang.org/x/sys https://github.com/golang/sys/compare/41f3e6584952bb034a481797859f6ab34b6803bd...d455e41777fca6e8a5a79e34a14b8368bc11d9ba
- https://github.com/docker/swarmkit/compare/ebfb0aa1118ebfd35a224d72a5d337ce0addd907...415dc72789e2b733ea884f09188c286ca187d8ec
- https://github.com/tonistiigi/fsutil/compare/2862f6bc5ac9b97124e552a5c108230b38a1b0ca...1ec1983587cde7e8ac2978e354ff5360af622464
- https://github.com/gogo/protobuf/compare/v1.1.1...v1.2.0
- https://github.com/golang/protobuf/compare/v1.1.0...v1.2.0
- https://github.com/opencontainers/runc/compare/12f6a991201fdb8f82579582d5e00e28fba06d0a...2b18fe1d885ee5083ef9f0838fee39b62d653e30
- https://github.com/containerd/containerd/compare/e6b3f5632f50dbc4e9cb6288d911bf4f5e95b18e...a15b6e2097c48b632dbdc63254bad4c62b69e709

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @silvin-lubecki @vdemeester PTAL

File: cli/command/system/testdata/docker-info-badsec.json.golden [hunk 1]
```diff
@@ -1 +1 @@
...
...ibutableArtifactsCIDRs":null,"AllowNondistributableArtifactsHostnames":null,"InsecureRegistryCIDRs":["127.0.0.0/8"],"IndexConfigs":{"docker.io":{"Name":"docker.io","Mirrors":null,"Secure":true,"Official":true}},"Mirrors":null},"NCPU":2,"MemTotal":2097356800,"GenericResources":null,"DockerRootDir":"/var/lib/docker","HttpProxy":"","HttpsProxy":"","NoProxy":"","Name":"system-sample","Labels":["provider=digitalocean"],"ExperimentalBuild":false,"ServerVersion":"17.06.1-ce","ClusterStore":"","ClusterAdvertise":"","Runtimes":{"runc":{"path":"docker-runc"}},"DefaultRuntime":"runc","Swarm":{"NodeID":"","NodeAddr":"","LocalNodeState":"inactive","ControlAvailable":false,"Error":"","RemoteManagers":null},"LiveRestoreEnabled":false,"Isolation":"","InitBinary":"docker-init","ContainerdCommit":{"ID":"6e23458c129b551d5c9871e5174f6b1b7f6d1170","Expected":"6e23458c129b551d5c9871e5174f6b1b7f6d1170"},"RuncCommit":{"ID":"810190ceaa507aa2727d7ae6f4790c76ec150bd2","Expected":"810190ceaa507aa2727d7ae6f4790c76ec150bd2"},"InitCommit":{"ID":"949e6fa","Expected":"949e6fa"},"SecurityOptions":["foo="],"Warnings":null,"ServerErrors":["an error happened"],"ClientInfo":{"Debug":false,"Plugins":[],"Warnings":null}}
```
Comment (silvin-lubecki) [inline cli/command/system/testdata/docker-info-badsec.json.golden]: Not easy to review what changed 😅

File: vendor.conf [hunk 2]
```diff
@@ -4,7 +4,7 @@ github.com/asaskevich/govalidator f9ffefc3facfbe0caee3fea233cbb6e8208f4541
 github.com/Azure/go-ansiterm d6e3b3328b783f23731bc4d058875b0371ff8109
 github.com/beorn7/perks 3a771d992973f24aa725d07868b467d1ddfceafb
 github.com/containerd/console c12b1e7919c14469339a5d38f2f8ed9b64a9de23
-github.com/containerd/containerd e6b3f5632f50dbc4e9cb6288d911bf4f5e95b18e # v1.2.4
+github.com/containerd/containerd a15b6e2097c48b632dbdc63254bad4c62b69e709
```
Comment (silvin-lubecki) [inline vendor.conf]: Is it a version used transitively by moby/moby? Is there any associated release?

Comment (thaJeztah) [reply vendor.conf]: Had that discussion a couple of times as well; basically, the Go client of containerd does not have a stable API (see https://github.com/containerd/containerd/blob/master/RELEASES.md#public-api-stability), so using a version that's more recent than the runtime itself can be needed at times 

I still try to stick to tagged versions where possible, but in this case, some changes were needed for buildkit that are not yet in a release (perhaps a new release is done before 19.03 GA, in which case (possibly) we could pin to a tagged version)

**Pull request merged**